lib/internal/test_runner/reporter/spec.js

```diff
@@ -74,31 +100,18 @@ class SpecReporter extends Transform {
       case 'test:coverage':
         return getCoverageReport(indent(data.nesting), data.summary,
                                  reporterUnicodeSymbolMap['test:coverage'], colors.blue, true);
+      case 'test:summary':
+        // We report only the root test summary
+        if (data.file === undefined) {
+          return this.#formatFailedTestResults();
+        }
     }
   }
   _transform({ type, data }, encoding, callback) {
     callback(null, this.#handleEvent({ __proto__: null, type, data }));
   }
   _flush(callback) {
-    if (this.#failedTests.length === 0) {
-      callback(null, '');
-      return;
-    }
-    const results = [`\n${reporterColorMap['test:fail']}${reporterUnicodeSymbolMap['test:fail']}failing tests:${colors.white}\n`];
-    for (let i = 0; i < this.#failedTests.length; i++) {
-      const test = this.#failedTests[i];
-      const formattedErr = formatTestReport('test:fail', test);
-
-      if (test.file) {
-        const relPath = relative(this.#cwd, test.file);
-        const location = `test at ${relPath}:${test.line}:${test.column}`;
-
-        ArrayPrototypePush(results, location);
-      }
-
-      ArrayPrototypePush(results, formattedErr);
-    }
-    callback(null, ArrayPrototypeJoin(results, '\n'));
+    callback(null, this.#formatFailedTestResults());
```

Comment:
We shouldn't still need this, right?

Reply:
In non-watch mode, wouldn't it cause the errors to be printed here and for the summary event?

Reply:
nope as on each "print" we're cleaning `this.#failedTests = []; // Clean up the failed tests`.
But still we can remove it

Reply:
Ah ok. I guess the question is, "is this expected to do anything?" My thoughts are no, so we should probably remove it.

Reply:
I've just checked, and I was wrong.  

Removing `_flush` breaks the following tests:

<img width="518" alt="image" src="https://github.com/user-attachments/assets/7943c77b-55f0-48c9-a522-8109cdc3a1a1" />

Reply:
I agree!

Reply:
One more thing to note about my proposal in my previous comment - that change needs to happen in a semver-major and it also shouldn't happen until all release lines that **don't** support `test:summary` are end-of-life, as it would break userland reporters with no workaround. Fun.